src/car.rs

```diff
@@ -0,0 +1,83 @@
+use crate::error::Error;
+
+pub struct Car {
+    pub name: String,
+    pub distance: u32,
+}
+
+impl Car {
+    pub fn new(name: &str) -> Result<Car, Error> {
+        let name = name.trim();
+        let name_char_length = name.chars().count();
+        if name_char_length == 0 {
+            return Err(Error::BlankCarName);
+        }
+        if name_char_length > 5 {
+            return Err(Error::CarNameOverFiveCharacters);
+        }
```

Comment:
There is validation for the name of the Car, how about delegating the responsibility for this to the CarName struct? You can separate the test code for name.